src/components/v5/common/ActionSidebar/partials/Motions/utils.ts

```diff
@@ -0,0 +1,348 @@
+    id: `${displayName}.finalizeYaySideWon`,
+    defaultMessage:
+      'Action can now be finalized and stakes can be returned. {timestamp}',
+  },
+  finalizeNaySideWon: {
+    id: `${displayName}.finalizeNaySideWon`,
+    defaultMessage: 'Action failed stakes can be returned. {timestamp}',
+  },
+  finalizedAction: {
+    id: `${displayName}.finalizedAction`,
+    defaultMessage: 'Action was finalized. {timestamp}',
+  },
+});
+
+const formatMotionTooltipTimestamp = (timestamp: string | undefined | null) =>
+  timestamp
+    ? capitalizeFirstLetter(formatRelative(new Date(timestamp), new Date()))
+    : '';
+
+export const getStakingStepTooltipText = (
+  motionState: MotionState | undefined,
+  motionData: ColonyMotion | undefined | null,
+) => {
+  const {
+    motionStakes,
+    createdAt: motionCreatedAt,
+    motionStateHistory,
+  } = motionData || {};
+  const { percentage } = motionStakes || {};
+  const { naySideFullyStakedAt, yaySideFullyStakedAt, endedAt } =
+    motionStateHistory || {};
+  const { nay, yay } = percentage || {};
+
+  const formatttedMotionCreatedAt =
+    formatMotionTooltipTimestamp(motionCreatedAt);
+  const formattedNaySideFullyStakedAt =
+    formatMotionTooltipTimestamp(naySideFullyStakedAt);
+  const formattedYaySideFullyStakedAt =
+    formatMotionTooltipTimestamp(yaySideFullyStakedAt);
+  const formattedEndedAt = formatMotionTooltipTimestamp(endedAt);
+  const objectingStakesPercentageValue = Number(nay) || 0;
+  const supportingStakesPercentageValue = Number(yay) || 0;
+
+  const isFullyOpposed = objectingStakesPercentageValue === 100;
+  const isFullySupported = supportingStakesPercentageValue === 100;
+
+  const isFullyStaked = isFullyOpposed && isFullySupported;
+
+  if (motionState === MotionState.Staking) {
+    if (!(isFullyOpposed || isFullyStaked)) {
+      return formatText(MSG.stakingStarted, {
+        timestamp: formatttedMotionCreatedAt,
+      });
+    }
+
+    if (isFullyOpposed) {
+      return formatText(MSG.stakingFullyOpposed, {
+        timestamp: formattedNaySideFullyStakedAt,
+      });
+    }
+
+    if (isFullySupported) {
+      return formatText(MSG.stakingFullySupported, {
+        timestamp: formattedYaySideFullyStakedAt,
+      });
+    }
+  }
+
+  if (isFullyStaked) {
+    return formatText(MSG.stakingBothSidesOutcome, {
+      timestamp:
+        Number(naySideFullyStakedAt) > Number(yaySideFullyStakedAt)
+          ? formattedNaySideFullyStakedAt
+          : formattedYaySideFullyStakedAt,
+    });
+  }
+
+  if (isFullyOpposed) {
+    return formatText(MSG.stakingFullyOpposedOutcome, {
+      timestamp: formattedEndedAt,
+    });
+  }
+
+  if (isFullySupported) {
+    return formatText(MSG.stakingFullySupportedOutcome, {
+      timestamp: formattedEndedAt,
+    });
+  }
+
+  return formatText(MSG.stakingNoSidesOutcome, {
+    timestamp: formattedEndedAt,
+  });
+};
+
+export const getVotingStepTooltipText = (
+  motionState: MotionState | undefined = MotionState.Null,
```

Comment:
I mean this can't really be `undefined` since it has a default value of `MotionState.Null` if it's really undefined, so it should only be `MotionState`

I wonder why `tslint` didn't complain about it